### ISSUE TITLE
train_size referenced before assignment

### ISSUE THREAD

**Issue by bayethiernodiop** [mozilla/CorporaCreator#93]
Hello, when trying to use the create-corpora -d corpora -f out/clips.tsv command i have the following error UnboundLocalError: local variable 'train_size' referenced before assignment caused by at line 115.
Help please.

**Comment by bayethiernodiop** [comment]
after checking the code the bug is for a given langauge if there is no validated audio the train_size will not be intialized. the reason is that the len(validated is 0)

**Comment by kdavis-mozilla** [comment]
This was created to deal with the release clips.tsv and not a clips.tsv generated after the release. The release clips.tsv did not have any languages without validated clips.

That said, crashing, as it does now, or issuing a warning and exiting seems like the only reasonable behaviors in this situation.

What do you think?

**Comment by bayethiernodiop** [comment]
ah ok i see. i was thinking of ignoring validation set for language without validated recordings.

**Comment by bayethiernodiop** [comment]
and print a warning because it may be a problem for those deploying the common voice locally.

**Comment by kdavis-mozilla** [comment]
@Gregoor What would be you desired behavior for the scripts that create the Common Voice releases?

My gut says a catastrophic failure might be of more use in that situation as it will definitely be noticed while a "silent" error being printed might not be noticed and data sets would go out without data.

**Comment by Gregoor** [comment]
My take would be that we should notice at some other point that we don't have validated data for a language. The bundler script also gathers stats, in part based on the corpora creator's result, which would show if a validated set is empty for a language.

**Comment by kdavis-mozilla** [comment]
@Gregoor I guess my question is this: If the stats are gathered, will they be reliably looked at?

There are 96 languages in [pontoon](https://pontoon.mozilla.org/projects/common-voice/) so it's not so unlikely that if 96, or more, languages are released one stat will be overlooked.

However, if the bundler script doesn't run to completion, then no stats are there, and that _will_ be noticed.

**Comment by Gregoor** [comment]
Yeah that's a fair point, probably not. But I think the bundler would be a better place to act on this. And I guess we have to decide how we'd wanna act on it (maybe exclude the language from release?).

**Comment by bayethiernodiop** [comment]
i think exclude it from release is a good choice.
